lldb/source/Symbol/Type.cpp

```diff
@@ -153,19 +127,36 @@ void TypeQuery::SetLanguages(LanguageSet languages) {
 
 bool TypeQuery::ContextMatches(
     llvm::ArrayRef<CompilerContext> context_chain) const {
-  if (GetExactMatch() || context_chain.size() == m_context.size())
-    return ::contextMatches(context_chain, m_context);
+  auto ctx = context_chain.rbegin(), ctx_end = context_chain.rend();
+  for (auto pat = m_context.rbegin(), pat_end = m_context.rend();
```

Comment:
Could this loop be just a `std::search`?

Reply:
With the modules out of the way, I guess it could be, but then I'd have to rewrite it again to this form for anonymous namespaces.